src/batch.rs

```diff
@@ -0,0 +1,61 @@
+//! # Batch
+//!
+//! This module contains types for batch operations. A batch operation is a
+//! collection of write operations (puts and/or deletes) that are applied
+//! atomically to the database.
+
+/// A batch of write operations (puts and/or deletes). All operations in the
+/// batch are applied atomically to the database. If multiple operations appear
+/// for a a single key, the last operation will be applied. The others will be
+/// dropped.
```

Comment:
do we ever want to support partial/merge updates? something like https://github.com/facebook/rocksdb/wiki/merge-operator - I don't imagine this throws any wrench in the plan here, but thought it might be interesting how it works in tandem with atomic batches

Reply:
Not sure. I think this could be implemented as a transaction once we support them. Since we're doing light weight optimistic locks, I think it should be OK to just use that, right?